### ISSUE TITLE
Runloop debounce very slow in Chrome

### ISSUE THREAD

**Issue by rmmmp** [emberjs/ember.js#16645]
So I've attached a scroll listener to the `window` of the page. Naturally, I'd like to wrap that under a `debounce` so the action to be done doesn't fire crazily. What I've found out is that the `debounce` for a scroll listener when attached to `window` is very slow **for Chrome**. I've set a timer of 150ms but it seems to use 1000+ms. The same issue doesn't happen in Firefox and Edge. Also, attaching the scroll listener to any element other than `window` seems to work fine in Chrome.

Since the issue happens in `window`, I can't replicate it in Ember Twiddle so I've put up a [repo](https://github.com/rmmmp/ember-debounce-bug) to show the bug. It uses Ember ~3.1.0.

I don't believe this is a Chrome issue as I've tried removing the `debounce` and everything fires correctly. No delay or anything. So this leads me to think that it's a `debounce` bug.

**Comment by Turbo87** [comment]
@rmmmp I would assume that `debounce()` is using `setTimeout()` under the hood, which Chrome itself throttles in certain cases (e.g. tab in the background, scrolling, etc.). The guarantee that `debounce()` gives is that it will never be fired twice in the same time interval, but due to the `setTimeout()` behavior it might take a little while before the next time it is fired.

tl;dr this is outside of our control if the browser is throttling `setTimeout()` 😞

**Comment by rmmmp** [comment]
Closing with a sad face
